website/content/specification/_index.md

```diff
@@ -1,50 +1,1320 @@
+{{</callout>}}
+
+### Naming and `<use-name>`
+
+The optional `<use-name>` changes the *effective name* to use for the information element in a data model.
+
+The `<use-name>` element is optional and MAY only occur once.
+
+By default the *effective name* of the information element in a data model is taken from the `@name` attribute. The `<use-name>` value overrides this behavior.
+
+{{<callout>}}
+Use of a `<use-name>` frees the module maintainer allowing them to use a sensible `@name` for the definition, while providing control over the name used in an instance of the definition in a data model.
+{{</callout>}}
+
+The first matching condition determines the *effective name* for the definition:
+
+1. A `<use-name>` is provided on the definition. The *effective name* is the value of the `<use-name>` element on the definition.
+1. No `<use-name>` is provided on the definition. The *effective name* is the value of the `@name` attribute on the definition.
+
+For example:
+
+```xml {linenos=table,hl_lines=[2]}
+<define-flag name="flag-a">
+  <use-name>flag-b</use-name>
+</define-flag>
+```
+
+In the example above, the *effective name* of the definition is `flag-b`. If the `<use-name>` was omitted, the *effective name* would be `flag-a`.
+
+The following content is valid to the model above.
+
+{{< tabs JSON YAML XML >}}
+{{% tab %}}
+```json
+{
+  "field": {
+    "flag-b": "value"
+  }
+}
+```
+{{% /tab %}}
+{{% tab %}}
+```yaml
+---
+field:
+  flag-b: "value"
+```
+{{% /tab %}}
+{{% tab %}}
+```xml
+<field flag-b="value"/>
+```
+{{% /tab %}}
+{{% /tabs %}}
+
+### `<remarks>`
+
+The optional `<remarks>` element provides a place to add notes related to the use of the definition. Remarks can be used to clarify the semantics of the definition in specific conditions, or to better describe how the definition may be more fully utilized within a model. 
+
+The `<remarks>` element is optional and may occur multiple times.
+
+### `<example>`
+
+The optional `<example>` element is used to provide inline examples, which are intended to illustrate the use the information element implemented by the definition. Examples are provided in XML, which can then be automatically converted into other formats.
+
+The `example` element is optional and may occur multiple times.
+
+## top-level `<define-flag>`
+
+A flag definition, represented by the `<define-flag>` element, is used to declare a reusable [flag](/specification/terminology/#flag) within a Metaschema module.
+
+A flag definition provides the means to implement a simple, named [*information element*](/specification/terminology/#information-element) with a value.
+
+{{<callout>}}
+Flag definitions are the primary leaf nodes in a Metaschema-based model. Flags are intended to represent granular particles of identifying and qualifying information.
```

Comment:
I find this language a bit cumbersome, but also I am not completely happy with what I wrote.

I'd rather not discuss fields or field values here if possible. I'd also like to avoid introducing parallel terminology such as leaf nodes / terminal nodes.

Perhaps we could collaborate on something?

Reply:
Point taken about not confusing with ideas better explained elsewhere.

I ended up going there because of the ambiguity raised by 'primary leaf node' - if primary, what's secondary and do we have tertiary as well? More precise might be to say that they typically represent the majority of leaf nodes? Part of the reason for 'primary' is that there is only one other kind -- but arguably of the two, a field value is 'primary' since it typically shows a data element of some kind.

We could start by striking 'primary' and then seeing how to communicate the idea better?

Reply:
This content has been rewritten, so this thread is OBE.